src/main/java/teammates/ui/webapi/QueryLogsAction.java

```diff
@@ -76,10 +98,42 @@ public JsonResult execute() {
                 .build();
         try {
             QueryLogsResults queryResults = logsProcessor.queryLogs(queryLogsParams);
+            removeSensitiveFields(queryResults);
             GeneralLogsData generalLogsData = new GeneralLogsData(queryResults);
             return new JsonResult(generalLogsData);
         } catch (LogServiceException e) {
             return new JsonResult(e.getMessage(), HttpStatus.SC_INTERNAL_SERVER_ERROR);
         }
     }
+
+    private void removeSensitiveFields(QueryLogsResults queryResults) {
+        if (userInfo.isAdmin) {
+            return;
+        }
+
+        for (GeneralLogEntry logEntry : queryResults.getLogEntries()) {
+            if (logEntry.getDetails() != null) {
+                Map<String, Object> details = logEntry.getDetails();
+                // Always remove requestParams, requestHeaders and userInfo for non-admin maintainers
```

Comment:
No need to repeat `non-admin maintainers` so many times. It's quite clear from the code flow that everything past the early return is meant for non-admins.

Reply:
Ok. It is updated.